### PR TITLE
Fix fails on ES6 method syntax in ClientFunction arguments (closes #1279)

### LINKED ISSUE #1279: TestCafe fails with error if ClientFunction argument contains ES6 method syntax
### Are you requesting a feature or reporting a bug?
bug

### What is the current behavior?
TestCafe fails if ClientFunction argument contains ES6 method syntax

### What is the expected behavior?
TestCafe shouldn't fail

### How would you reproduce the current behavior (if this is a bug)?
Run test

#### Provide the test code and the tested page URL (if applicable)

Test code

```js
import { ClientFunction } from 'testcafe';

fixture `fixture`;

var fn = ClientFunction(() => {});

test('test', async () => {
    await fn({
        es6Method () {}
    });
});
```

### PR DESCRIPTION
**Pull request by inikulin**
\cc @kirovboris @AndreyBelym @AlexanderMoskovkin

### CODE REVIEW

File: src/compiler/es-next/compile-client-function.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ import { ClientFunctionAPIError } from '../../errors/runtime';
 import MESSAGE from '../../errors/runtime/message';
 
 const ANONYMOUS_FN_RE                = /^function\*?\s*\(/;
+const ES6_OBJ_METHOD_NAME_RE         = /^([A-Za-z\*\$]+)\s*\(/;
```
Comment (kirovboris) [inline src/compiler/es-next/compile-client-function.js]: It seems this regex too weak for function name, e.g. you can use '_' or numbers.

Comment (inikulin) [inline src/compiler/es-next/compile-client-function.js]: good catch

**Comment by kirovboris** [comment]
@testcafe-build-bot \retest

**Comment by kirovboris** [comment]
@testcafe-build-bot \retest

**Pull request merged**